lrpar/src/lib/builder.rs

```diff
@@ -152,10 +154,30 @@ where StorageT: 'static + Debug + Hash + PrimInt + Serialize + TypeName + Unsign
         let rule_ids = grm.terms_map().iter()
                                       .map(|(&n, &i)| (n.to_owned(), i.as_storaget()))
                                       .collect::<HashMap<_, _>>();
+        let imc = self.gen_ids_map_cache(&grm);
+
+        // We don't need to go through the full rigmarole of generating an output file if all of
+        // the following are true: the output file exists; it is newer than the input file; and the
+        // rule IDs map cache hasn't changed. The last of these might be surprising, but it's
+        // vital: we don't know what the IDs map might be from one run to the next, and it might
+        // change for reasons beyond lrpar's control. If it does change, that means that the lexer
+        // and lrpar would get out of sync, so we have to play it safe and regenerate in such cases.
+        if let Ok(ref inmd) = fs::metadata(&inp) {
```

Comment:
It seems odd to me to have so many nested `if`s rather than using `&&`

Reply:
You can't chain `if let`s :/ Which is very annoying.